### PR TITLE
Add "Check Configuration" and addons support to istio adapter

### LINKED ISSUE #160: Operation: "Check configuration" failing in v0.4.10
**Description**
When Istio is **not deployed**, running "check configuration" fails with the error below.

The user receives no feedback in Meshery UI.

**Logs**
```
meshery-istio_1     | E1205 00:02:54.003782       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.8/tools/cache/reflector.go:125: Failed to list *v1beta1.VirtualService: the server could not find the requested resource (get virtualservices.networking.istio.io)
```

**Enviroment:**
 - Version: v0.4.10

### PR DESCRIPTION
**Pull request by utkarsh-pro**
**Description**
This PR adds several features to the istio adapter:
1. Check Configuration (Istio Vet)
2. Istio Addons
    1. Grafana Dashboard
    2. jaeger Dashboard
    3. Kiali Dashboard
    4. Prometheus Monitoring
    5. Zipkin Dashboard

This PR may also fix #160 after a successful release.
This PR also modifies the dockerfile to adapt the recent changes in the adapter. 

Tests performed (both inside and outside the container) :
- [x] Install/Uninstall all the addons
- [x] Run "check configuration"
- [x] Run `golangci-lint` `gosec ./... -exclude=G301,G304,G107,G101,G110` `errcheck ./...`

**Notes for Reviewers**
https://github.com/utkarsh-pro/meshery-istio/blob/utkarsh-pro%2Ffeature%2Faddons-support/istio/addons.go#L53 was added specifically to capture Kiali's dashboard deletion error which probably arises due to this [issue](https://github.com/kiali/kiali/issues/3112)

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: internal/config/operations.go [hunk 1]
```diff
@@ -27,5 +27,35 @@ func getOperations(dev adapter.Operations) adapter.Operations {
...
+	}
+
+	dev[ZipkinAddon] = &adapter.Operation{
+		Type:        int32(meshes.OpCategory_CONFIGURE),
+		Description: "Zipkin Dashboard",
+	}
+
+	dev[IstioVetOpertation] = &adapter.Operation{
+		Type:        int32(meshes.OpCategory_VALIDATE),
+		Description: "Check Configuration",
```
Comment (leecalcote) [inline internal/config/operations.go:57]:
```suggestion
		Description: "Analyze Running Configuration",
```

File: istio/istio.go [hunk 2]
```diff
@@ -113,6 +115,43 @@ func (istio *Istio) ApplyOperation(ctx context.Context, opReq adapter.OperationR
...
+
+			if err != nil {
+				e.Summary = fmt.Sprintf("Error while %sing %s", operation, opReq.OperationName)
+				e.Details = err.Error()
+				hh.StreamErr(e, err)
+				return
+			}
+			ee.Summary = fmt.Sprintf("Succesfully %sed %s", operation, opReq.OperationName)
+			ee.Details = fmt.Sprintf("Succesfully %sed %s from the %s namespace", operation, opReq.OperationName, opReq.Namespace)
+			hh.StreamInfo(e)
```
Comment (leecalcote) [inline istio/istio.go:134]: If we're able to pass along the URL to the new endpoint created for the user to access the add-on... that'd be helpful. 

Not as helpful, though, as adding Kiali and Jaeger to the list of things MeshSync v1 discovers.

**Review by leecalcote — APPROVED** [review]
Thank you, @utkarsh-pro.

File: istio/istio.go (hunk 2 above)
Comment (utkarsh-pro) [reply istio/istio.go:134]: @leecalcote, we have a function in meshkit called `GetServiceEndpoint` which returns the address for the service. We can achieve the same thing in the MeshSync v1 by tweaking it a bit. But, both kiali and jaeger addon manifest only gives us a cluster IP based endpoint by default. 
Would that be fine? 

Also, we have a [generic expose function](https://github.com/layer5io/meshery-linkerd/pull/160) in linkerd adapter which can expose these services and give them a `nodeport`/`loadbalancer` based address, so we can copy that function here or maybe move it to meshkit, if we want to expose these services too.

Another appoach could be to modify the default addon manifest and change service type to `NodePort`/`LoadBalancer`, I am not sure if tweaking the original manifest would be a good solution or not :sweat_smile: .

Any suggestions?

Comment (leecalcote) [reply istio/istio.go:134]: Fantastic list. Let's get the `expose` function bubble up to meshkit and then back down across each of the adapters. 👍

**Pull request merged**